flat_mutation_reader_v2.hh

```diff
@@ -881,6 +881,27 @@ make_flat_mutation_reader_from_fragments(schema_ptr, reader_permit, std::deque<m
 flat_mutation_reader_v2
 make_flat_mutation_reader_from_fragments(schema_ptr, reader_permit, std::deque<mutation_fragment_v2>, const dht::partition_range& pr, const query::partition_slice& slice);
 
+// Calls the consumer for each element of the reader's stream until end of stream
+// is reached or the consumer requests iteration to stop by returning stop_iteration::yes.
+// The consumer should accept mutation as the argument and return stop_iteration.
+// The returned future<> resolves when consumption ends.
+template <typename Consumer>
+inline
+future<> consume_partitions(flat_mutation_reader_v2& reader, Consumer consumer) {
+    static_assert(std::is_same<future<stop_iteration>, futurize_t<std::result_of_t<Consumer(mutation&&)>>>::value, "bad Consumer signature");
```

Comment:
We need to replace this with a concept. Can be a follow-up (I know this is a clone of the v1 `consume_partitions()` which also doesn't use concepts).